File: 45-jump-game-ii/jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int n = nums.length;
        if (n < 2)
            return 0;

        int maxReach = nums[0];
        int t = nums[0];
        int steps = 1;

        for (int i = 1; i < n; i++) {
            if (i > maxReach)
                return -1; // unreachable
            if (i > t) {
                steps++;
                t = maxReach;
            }
            maxReach = Math.max(maxReach, i + nums[i]);
        }

        return steps;
    }
}

File: 45-jump-game-ii/jump-game-ii.go
func jump(nums []int) int {
	maxLen := 0
    tempMax:= 0
    steps:=0
	for i := 0; i < len(nums) && i <= maxLen; i++ {
		if i+nums[i] > tempMax {
			tempMax = i + nums[i]
		}
        if i == maxLen && i!=len(nums)-1{
            maxLen=tempMax
            steps++
        }

	}

    return steps
}